### ISSUE TITLE
Usage in the browser

### ISSUE THREAD

**Issue by mrf-ipt** [rdfjs/N3.js#205]
Hi, I'm struggeling to use N3 in the browser. I want to use it in an Angular project, which I guess uses Webpack.

I tried just installing (`npm i --save n3 && npm i --save-dev @types/n3`) the package and then importing it `import { Parser } from 'n3'` but that throws errors because the `stream` module is not found. 

I tried installing it, and then generating a Webpack bundle in `./node_modules/n3`. I can include it fine but at runtime get an error that the imported symbol `_vendor_n3__WEBPACK_IMPORTED_MODULE_3___default.a.Parser` is not a constructor. So I guess there is a miss match between the exports of the Webpack bundle and the types?

So my question is, what is the correct way to include the N3 library in a Angular/Webpack/Typescript project?

**Comment by RubenVerborgh** [comment]
> because the `stream` module is not found

Weird, that is a Node.js built-in module.

> So my question is, what is the correct way to include the N3 library in a Angular/Webpack/Typescript project?

For webpack, should just be normal really. We've done it many times.

Do you have a config for me?

**Comment by mrf-ipt** [comment]
I think I figured it out. The problem is that Angular CLI does not expose the Webpack config. It has a fixed default config, which does not include the node module shims, especially not the stream module. It 's more or less discussed here: https://github.com/angular/angular-cli/issues/9827

The solution for me right now was to install and use the `@angular-builders/custom-webpack` package. Then I can customize the Webpack configuration. I added 
```js
module.exports = {
    node: {
        stream: true
    }
}
```
and now it works. This configures Webpack to provide a polyfill for the `stream` module.

In the long term it might be a good idea to detect whether the module is running in a Node.js environment and only then include the `stream` module.
